File: SproInstall/SocketPro/tutorial/CSharp/SampleThree/SharedItem/Shared.cs
using System;
using SocketProAdapter;

namespace SampleThreeShared
{
	public class CTestItem : IUSerializer
	{
		public CTestItem()
		{
			m_dt = System.DateTime.Now;
		}
		public void SaveTo(CUQueue UQueue)
		{
            //make sure that both native and .NET codes are compatible to each of other
			object obj = m_dt;
			UQueue.Push(obj); //save datetime as native VARIANT datetime
			UQueue.Push(m_lData);
            UQueue.Save(m_strUID);
		}
		public int LoadFrom(CUQueue UQueue)
		{
			int nLen;
			object obj = null;
            //make sure that both native and .NET codes are compatible to each of other
			nLen = UQueue.Pop(out obj);
			m_dt = (DateTime)obj;
			nLen += UQueue.Pop(out m_lData);
            nLen += UQueue.Load(out m_strUID);
			return nLen;
		}
		public DateTime	m_dt;
		public long		m_lData;
		public string	m_strUID;
	}
}

File: SproInstall/SocketPro/tutorial/VBNet/SampleThree/SharedItem/Shared.vb
Imports SocketProAdapter

Namespace SampleThreeShared
	Public Class CTestItem
        Implements IUSerializer
		Public Sub New()
			m_dt = Date.Now
		End Sub
        Public Sub SaveTo(ByVal UQueue As CUQueue) Implements IUSerializer.SaveTo
            'make sure that both native and .NET codes are compatible to each of other
            Dim obj As Object = m_dt
            UQueue.Push(obj) 'save datetime as native VARIANT datetime
            UQueue.Push(m_lData)
            UQueue.Save(m_strUID)
        End Sub
        Public Function LoadFrom(ByVal UQueue As CUQueue) As Integer Implements IUSerializer.LoadFrom
            Dim nLen As Integer
            Dim obj As Object = Nothing

            'make sure that both native and .NET codes are compatible to each of other
            nLen = UQueue.Pop(obj)
            m_dt = CDate(obj)
            nLen += UQueue.Pop(m_lData)
            nLen += UQueue.Load(m_strUID)
            Return nLen
        End Function
		Public m_dt As Date
		Public m_lData As Long
		Public m_strUID As String
	End Class
End Namespace